azure-iot-device/azure/iot/device/provisioning/provisioning_device_client.py

```diff
@@ -41,14 +41,17 @@ def register(self):
         """
         logger.info("Registering with Provisioning Service...")
         register_complete = Event()
+        registration_result = [1]
```

Comment:
>registration_result = [1] [](start = 8, length = 25)

1) Why are you using a list instead of a regular variable? Local variables should be settable via the callback
2) Even if you were to use a list, initializing a list with a value of `1` and then overwriting that value by indexing is not a typical pattern at all.